docs/user-guide/creating-application/secrets.md

```diff
@@ -50,25 +39,102 @@ File permission will be provide at the configmap level not on the each key of th
 
 To update secrets, click on the secret you wish to update.
 
-![](../../.gitbook/assets/updating_secrets.png)
+![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/secrets/creating-applications-secrets-6.jpg)
 
 Click on `Update Secret` to update your secret.
 
 ## Delete Secret
 
 You can delete your secret. Click on your secret and click on the `delete sign` to delete your secret.
 
-![](../../.gitbook/assets/delete_secret%20%282%29.png)
+![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/secrets/creating-applications-secrets-7.jpg)
+
+## Data Types
+
+There are five Data types that you can use to save your secret.
+
+* **Kubernetes Secret**: The secret that you create using Devtron.
+* **Kubernetes External Secret**: The secret data of your application is fetched by Devtron externally. Then the Kubernetes External Secret is converted to Kubernetes Secret.
+* **AWS Secret Manager**: The secret data of your application is fetched from AWS Secret Manager and then converted to Kubernetes Secret from AWS Secret. 
+* **AWS System Manager**: The secret data for your application is fetched from AWS System Secret Manager and all the secrets stored in AWS System Manager are converted to Kubernetes Secret.
+* **Hashi Corp Vault**: The secret data for your application is fetched from Hashi Corp Vault and the secrets stored in Hashi Corp Vault are converted to Kubernetes Secret.
+
+*Note: The conversion of secrets from various data types to Kubernetes Secrets is done within Devtron and irrespective of the data type, after conversion, the Pods access `secrets` normally.*
+
+## External Secrets
+
+In some cases, it may be that you already have secrets for your application on some other sources and you want to use that on devtron. External secrets are fetched by devtron externally and then converted to kubernetes secrets. 
+
+### Kubernetes External Secret
+
+The secret that is already created and stored in the environment and being used by devtron externally is referred here as `Kubernetes External Secret`. For this option, devtron will not create any secret by itself but they can be used within the pods. Before adding secret from kubernetes external secret, please make sure that secret with the same name is present in the environment. To add secret from kubernetes external secret, follow the steps mentioned below:
+
+1. Navigate to `Secrets` of the application.
+2. Click on `Add Secret` to add a new secret.
+3. Select `Kubernetes External Secret` from dropdown of `Data type`.
+4. Provide a name to your secret. Devtron will search secret in the environment with the same name that you mention here. 
+
+### AWS Secret Manager
+
+Before adding any external secrets on devtron, `kubernetes-external-secrets` must be installed on devtron. Kubernetes External Secrets allows you to use external secret management systems (e.g., AWS Secrets Manager) to securely add secrets in Kubernetes.
```

Comment:
...must be installed on the target cluster.

(e.g., AWS Secrets Manager, Hashicorp Vault etc)

Reply:
done